src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,175 @@
+        ]
+    }
+
+
+ariaLabelToAttr : AriaLabel -> Attribute msg
+ariaLabelToAttr label =
+    case label of
+        IdLabel l ->
+            Aria.labeledBy l
+
+        StringLabel l ->
+            Aria.label l
+
+
+{-| Type which represents the type of aria label which will be used
+`IdLabel` will point to an existing element id on the DOM
+`StringLabel` will be a label of the element
+-}
+type AriaLabel
+    = IdLabel String
+    | StringLabel String
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+`controls`: tabs control buttons
+`slides` container with the carousel contents
+-}
+viewWithTabControls :
+    { cfg
+        | selected : id
+        , panels :
+            List
+                { id : id
+                , slideHtml : Html msg
+                , tabControlHtml : Html Never
+                , idString : String
+                }
+        , focusAndSelect : { select : id, focus : Maybe String } -> msg
+        , tabControlStyles : Bool -> List Style
+        , tabControlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        }
+viewWithTabControls config =
+    let
+        buildTab tabConfig =
+            TabsInternal.fromList { id = tabConfig.id, idString = tabConfig.idString }
+                [ \tab -> { tab | panelView = tabConfig.slideHtml }
+                , \tab -> { tab | tabView = [ Html.map never tabConfig.tabControlHtml ] }
+                ]
+
+        { tabList, tabPanels } =
+            TabsInternal.views
+                { focusAndSelect = config.focusAndSelect
+                , selected = config.selected
+                , tabs = List.map buildTab config.panels
+                , tabStyles = always config.tabControlStyles
+                , tabListStyles = config.tabControlListStyles
+                }
+    in
+    { controls = tabList
+    , slides = tabPanels
+    }
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+`tabControls`: tabs control buttons
+`slides` container with the carousel contents
+`previousAndNextControls`: previous and next buttons element
+-}
+viewWithCombinedControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , tabControlHtml : Html Never
+            , idString : String
+            }
+    , focusAndSelect : { select : id, focus : Maybe String } -> msg
+    , tabControlStyles : Bool -> List Style
+    , tabControlListStyles : List Style
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    }
+    ->
+        { tabControls : Html msg
+        , previousAndNextControls : Html msg
+        , slides : Html msg
+        }
+viewWithCombinedControls config =
+    let
+        { controls, slides } =
+            viewWithTabControls config
+    in
+    { tabControls = controls
+    , slides = slides
+    , previousAndNextControls = div [] [ config.viewPreviousButton, config.viewNextButton ]
```

Comment:
🎨 there's a little inconsistency in that `viewWithPreviousAndNextControls`receives a list of styles for the prev/next container, but this one doesn't. But maybe we won't need to add one anyways as per my other comment of returning the buttons separately!